lms/services/lms_api.py

```diff
@@ -0,0 +1,19 @@
+from typing import Literal, Protocol, Self, TypedDict
+
+
+class LMSDocument(TypedDict):
```

Comment:
Went for "document" here. This can be a File | Folder but also (not in this PR) a page.